### PR TITLE
[Fix][MapTool] Circle with M value 

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description

Follow up #42812, #42986 and #43060

This PR fixes M support for circle map tool.

I will open two other PR for Ellipse and Quadrilateral.

### CODE REVIEW

**Review by rouault — COMMENTED** [review]
That's quite a lot of testing code that is more or less a copy&paste of the same file. Couldn't the 4 test files merged into a single one with some parametrization of the test cases to test each of the 4 variants (XY, XYM, XYZ, XYZM) ?

**Comment by lbartoletti** [comment]
> That's quite a lot of testing code that is more or less a copy&paste of the same file. Couldn't the 4 test files merged into a single one with some parametrization of the test cases to test each of the 4 variants (XY, XYM, XYZ, XYZM) ?

Yes, indeed, there is a lot of copy/paste. I also thought about trying to come up with something to parameterize the tests, but I didn't go through with it. Why? I noticed that some tests were not always readable when you add several layers, sometimes you don't know on which layers you really snap on, etc.
I really preferred to isolate each tests to have a readability even for non developers.
I've done that for other tests lately, but if you think I should do that again, I will.

**Comment by rouault** [comment]
> I noticed that some tests were not always readable when you add several layers, sometimes you don't know on which layers you really snap on, etc.

I guess you could have a generic class TestQgsMapToolCircleAbstract with some parametrization of the layer name, default values and expected result, and lightweight deriving classes TestQgsMapToolCircleXY, TestQgsMapToolCircleAbstractXYZ, TestQgsMapToolCircleAbstractXYM, TestQgsMapToolCircleAbstractXYZM

**Review by rouault — APPROVED** [review]

File: src/app/qgsmaptooladdcircle.cpp [hunk 1]
```diff
@@ -50,6 +50,13 @@ void QgsMapToolAddCircle::deactivate()
       lineString->addZValue( point.z() );
       break;
     }
+    if ( QgsWkbTypes::hasM( point.wkbType() ) &&
+         point.z() != defaultMValue() )
```
Comment (rouault) [inline src/app/qgsmaptooladdcircle.cpp:54]: shouldn't that be ?
```suggestion
         point.m() != defaultMValue() )
```
not caught by tests ?

Comment (lbartoletti) [reply src/app/qgsmaptooladdcircle.cpp:54]: Right!
Yes it is not tested for now (as for the Z). It will be tested soon after my others PR on M/Z editing.

Comment (lbartoletti) [reply src/app/qgsmaptooladdcircle.cpp]: Mmm, I should split up my tests and introduce the snapping tests for circle into this PR

**Comment by rouault** [comment]
Is this PR good to merge ?

**Comment by lbartoletti** [comment]
> Is this PR good to merge ?

Yes, please.

**Comment by lbartoletti** [comment]
Thanks @rouault

**Pull request merged**